src/lib/containers/QueryWrapper.tsx

```diff
@@ -99,140 +127,127 @@ export type LockedFacet = {
-
 /**
- * Class wraps around any Synapse views that are dependent on a query bundle
- * Those classes then take in as props:
- *
- * @class QueryWrapper
- * @extends {React.Component}
+ * Component that manages the state of a Synapse table query. Data can be accessed via QueryContext using
+ * either `useQueryContext` or `QueryContextConsumer`.
  */
-export default class QueryWrapper extends React.Component<
-  QueryWrapperProps,
-  QueryWrapperState
-> {
-  private componentIndex: number
-  constructor(props: QueryWrapperProps) {
-    super(props)
-    this.executeInitialQueryRequest = this.executeInitialQueryRequest.bind(this)
-    this.executeQueryRequest = this.executeQueryRequest.bind(this)
-    this.getLastQueryRequest = this.getLastQueryRequest.bind(this)
-    this.getNextPageOfData = this.getNextPageOfData.bind(this)
-    this.updateParentState = this.updateParentState.bind(this)
-    this.getInitQueryRequest = this.getInitQueryRequest.bind(this)
-    this.getSelectedColumns = this.getSelectedColumns.bind(this)
-    const showFacetVisualization = props.defaultShowFacetVisualization ?? true
-
-    this.state = {
-      data: undefined,
-      isLoading: true,
-      isLoadingNewData: true,
-      hasMoreData: true,
-      lastFacetSelection: {
-        columnName: '',
-        facetValue: '',
-        selector: '',
-      },
-      chartSelectionIndex: 0,
-      isAllFilterSelectedForFacet: {},
-      loadNowStarted: false,
-      lastQueryRequest: cloneDeep(this.props.initQueryRequest!),
-      topLevelControlsState: {
-        showColumnFilter: true,
-        showFacetFilter: true,
-        showFacetVisualization,
-        showSearchBar: false,
-        showDownloadConfirmation: false,
-        showColumnSelectDropdown: false,
-        showSqlEditor: false,
-      },
-      isColumnSelected: [],
-      selectedRowIndices: [],
-      isFacetsAvailable: true,
-      error: undefined,
+export function QueryWrapper(props: QueryWrapperProps) {
+  const { initQueryRequest, onQueryChange, onQueryResultBundleChange } = props
+  const [lastQueryRequest, setLastQueryRequest] =
+    useState<QueryBundleRequest>(initQueryRequest)
+  const {
+    data: infiniteData,
+    hasNextPage,
+    fetchPreviousPage,
+    fetchNextPage,
+    isFetchingNextPage,
+    isLoading: queryIsLoading,
+    refetch,
+    error,
+    isPreviousData,
+    remove,
+  } = useInfiniteQueryResultBundle(lastQueryRequest, {
+    // We use `keepPreviousData` because we don't want to clear out the current data when the query is modified via the UI
+    keepPreviousData: true,
+  })
+
+  // Indicate if we're fetching data for the first time (queryIsLoading) or if we're fetching data for a brand new query (isPreviousData)
+  const isLoadingNewBundle = queryIsLoading || isPreviousData
+
+  const entityId = parseEntityIdFromSqlStatement(lastQueryRequest.query.sql)
+
+  const { data: entity } = useGetEntity(entityId)
```

Comment:
Some components use the Entity object, e.g. SynapseTable needs it to see if it's a File View. We will just always fetch it and include it in the QueryContext.